src/api.js

```diff
@@ -17,207 +17,221 @@ import currency from './currency';
 import functions from './functions';
 import * as utils from './utils';
 
-const options = {
-  store: null,
-  key: null,
-  url: null,
-  useCamelCase: null,
-  previewContent: null,
-};
-
-const api = {
-  version: '__VERSION__',
-  options,
-  request,
-
-  init(store, key, opt = {}) {
-    options.key = key;
-    options.store = store;
-    options.url = opt.url
-      ? utils.trimEnd(opt.url)
-      : `https://${store}.swell.store`;
-    options.vaultUrl = opt.vaultUrl
-      ? utils.trimEnd(opt.vaultUrl)
-      : `https://vault.schema.io`;
-    options.timeout = (opt.timeout && parseInt(opt.timeout, 10)) || 20000;
-    options.useCamelCase = opt.useCamelCase || false;
-    options.previewContent = opt.previewContent || false;
-    options.session = opt.session;
-    options.locale = opt.locale;
-    options.currency = opt.currency;
-    options.api = api;
-    options.getCookie = opt.getCookie || getCookie;
-    options.setCookie = opt.setCookie || setCookie;
-    options.getCart = opt.getCart;
-    options.updateCart = opt.updateCart;
-    utils.setOptions(options);
-  },
-
-  // Backward compatibility
-  auth(...args) {
-    return this.init(...args);
-  },
-
-  get(url, query) {
-    return request('get', url, query);
-  },
+function swell(initStore = undefined, initKey, initOpt = {}) {
+  const options = {
+    store: null,
+    key: null,
+    url: null,
+    useCamelCase: null,
+    previewContent: null,
+  };
 
-  put(url, data) {
-    return request('put', url, data);
-  },
+  const api = {};
 
-  post(url, data) {
-    return request('post', url, data);
-  },
+  Object.assign(api, {
```

Comment:
Instead of Object.assign, it might be better to convert the API client into a class.  Something like:

```
class ApiClient {
  constructor(store, publicKey, options = {}) {
    this.options = prepareOptions(...);
  }

  get(url, query) {
    return this.request('get', url, query);
  }

  ...
}
```

Reply:
A class makes much more sense, but I wanted to avoid doing much more refactoring until the next major version. For now this provides the same functional approach with as few changes as possible.